### PR TITLE
Vaults lack of slippage protection

### PR DESCRIPTION
**Pull request by DaigaroCota**

### CODE REVIEW

File: packages/protocol/src/abstracts/BaseVault.sol [hunk 1]
```diff
@@ -34,6 +34,10 @@ abstract contract BaseVault is ERC20, SystemAccessControl, PausableVault, VaultP
   error BaseVault__redeem_invalidInput();
   error BaseVault__setter_invalidInput();
   error BaseVault__checkRebalanceFee_excessFee();
+  error BaseVault__depositSlippageProtection();
+  error BaseVault__mintSlippageProtection();
+  error BaseVault__withdrawSlippageProtection();
+  error BaseVault__redeemSlippageProtection();
```
Comment (brozorec) [inline packages/protocol/src/abstracts/BaseVault.sol:40]: Rename according to the convention. I'd suggest for the error description to use "slippageTooHigh", for ex. `BaseVault__deposit_slippageTooHigh()`

File: packages/protocol/src/vaults/borrowing/BorrowingVault.sol [hunk 2]
```diff
@@ -46,6 +46,8 @@ contract BorrowingVault is BaseVault {
   error BorrowingVault__rebalance_invalidProvider();
   error BorrowingVault__rebalance_invalidFlasher();
   error BorrowingVault__checkFee_excessFee();
+  error BorrowingVault__borrowSlippageProtection();
+  error BorrowingVault__paybackSlippageProtection();
```
Comment (brozorec) [inline packages/protocol/src/vaults/borrowing/BorrowingVault.sol:50]: same comment as for the errors in "BaseVault"

**Review by brozorec — CHANGES_REQUESTED** [review]
This PR deals only with slippage protection on a vault level. Should we extend it to the Router too (maybe in another PR)?

**Comment by DaigaroCota** [comment]
> This PR deals only with slippage protection on a vault level. Should we extend it to the Router too (maybe in another PR)?

I recommend that we deal with router-swap slippage separately because is a complete different approach. I even think the slippage in the router should be implemented in the Swapper-types contract.

File: packages/protocol/src/vaults/borrowing/BorrowingVault.sol (hunk 2 above)
Comment (DaigaroCota) [reply packages/protocol/src/vaults/borrowing/BorrowingVault.sol]: fixed as suggested.

File: packages/protocol/src/abstracts/BaseVault.sol (hunk 1 above)
Comment (DaigaroCota) [reply packages/protocol/src/abstracts/BaseVault.sol]: fixed as suggested.

**Review by brozorec — APPROVED** [review]
:+1:

**Pull request merged**